docs/fastdds/ros2/Discovery_service/ros2_discovery_service.rst

```diff
@@ -0,0 +1,331 @@
+between them.
+
+We are going to execute an example with two different servers. The following image shows an
+schema of the architecture desired:
+
+.. image:: ds_partition_example.png
+    :align: center
+
+With this schema we are going to see how *Listener 1* is connected to *Talker 1* and *Talker 2*, but
+*Listener 2* will only listen *Talker 1*.
+
+We run the first server listening in localhost in default port 11811.
+
+.. code-block:: console
+
+    fastdds discovery -i 0 -l 127.0.0.1 -p 11811
+
+In another terminal we run the second server listening in localhost in port 11888 (random port).
+
+.. code-block:: console
+
+    fastdds discovery -i 1 -l 127.0.0.1 -p 11888
+
+Now we run in different terminals each of the nodes we want to run. Using the *env var*
+``ROS_DISCOVERY_SERVER`` we decide
+which server they are connected to. Be aware that the
+`ids must match <https://fast-dds.docs.eprosima.com/en/latest/fastdds/env_vars/env_vars.html>`__.
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811;127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_1
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER="127.0.0.1:11811"
+    ros2 run demo_nodes_cpp talker --ros-args --remap __node:=talker_2
+
+.. code-block:: console
+
+    export ROS_DISCOVERY_SERVER=";127.0.0.1:11888"
+    ros2 run demo_nodes_cpp listener --ros-args --remap __node:=listener_2
+
+We should see how *Listener 1* is receiving double messages while *Listener 2* is in a different
+partition of *Talker 2* and so it does not listen him.
+
+.. note::
+
+    Once two endpoints know each other, they do not need the server network between them to
+    listen to each other messages.
+
+
+
+Compare Discovery Service with Simple Discovery
+-----------------------------------------------
+
+In order to compare the ROS2 execution using *Simple Discovery* or *Discovery Service*,
+we would provide two scripts that execute a talker and many listeners and analyze the network traffic during this time.
+
+However, this functionality are references for advance purpose
+and we are going to leave its studying to the user.
+
+*Throw many listeners in simple and service mode to get traffic info:*
+* :download:`bash network traffic generator <generate_discovery_packages.bash>`
+
+*Analyze tcpdump traces and generates the graph:*
+* :download:`python3 graph generator <discovery_packets.py>`
+
+To successfully execute these scripts,
+first we must run the bash script with the argument to the *setup* path to source ROS2.
+This will generate the traffic trace for simple discovery.
+Executing the same script with second argument ``SERVER`` it will generates the trace for service discovery.
```

Comment:
Examples of the commands to execute the scripts would help a lot understanding what we mean here.